src/rust/catpowder/win/runtime.rs

```diff
@@ -126,19 +141,23 @@ impl PhysicalLayer for SharedCatpowderRuntime {
         let mut ret: ArrayVec<DemiBuffer, RECEIVE_BATCH_SIZE> = ArrayVec::new();
         let mut idx: u32 = 0;
 
-        if self.0.borrow_mut().rx.reserve_rx(Self::RING_LENGTH, &mut idx) == Self::RING_LENGTH {
-            let xdp_buffer: XdpBuffer = self.0.borrow().rx.get_buffer(idx);
-            let out: Vec<u8> = xdp_buffer.into();
-
-            let dbuf: DemiBuffer = DemiBuffer::from_slice(&out)?;
+        for rx in self.0.borrow_mut().rx.iter_mut() {
+            if rx.reserve_rx(Self::RING_LENGTH, &mut idx) == Self::RING_LENGTH {
```

Comment:
We can possibly reducing nesting by inverting this condition.

Reply:
This happens per ring for all N rings, so the order cannot be inverted.